crates/sargon/src/types/hd_signature.rs

```diff
@@ -48,6 +71,34 @@ impl<ID: SignableID> HDSignature<ID> {
     }
 }
 
+impl<ID: SignableID> From<&HDSignature<ID>> for SignatureWithPublicKey {
+    fn from(value: &HDSignature<ID>) -> Self {
+        let sig = value.signature;
```

Comment:
Ah I think instead of this you can change the `signature: Signature` inside of HDSignature` to contain `signature_with_publickey: SignatureWithPublicKey` and add **one extra check** in it, to validate that the publickey of the `signature_with_publickey` matches the `input`? Then this whole `From` can be removed and replaced by `hd_signature.signature_with_publickey` (`.clone()`) - feels better?